core/consensus/common/src/select_chain.rs

```diff
@@ -42,13 +36,11 @@ pub trait SelectChain<Block: BlockT>: Sync + Send + Clone {
 	/// best chain to author new blocks upon and probably finalize.
 	fn best_chain(&self) -> Result<<Block as BlockT>::Header, Error>;
 
-	/// Get the best ancestor of `target_hash` that we should attempt
-	/// to finalize next.
-	fn finality_target(
+	/// Get the best block in the fork containing `target_hash`, if any.
+	fn best_containing<'a>(
```

Comment:
That's not the goal of `finality_target` -- the previous goal was to return the highest ancestor block which was _suitable to finalize_. This is typically the identity function, but for e.g. Polkadot we need to have additional voting rules based on what erasure-coded pieces we've seen.